code/modules/mining/machine_processing.dm

```diff
@@ -141,6 +148,14 @@
 	QDEL_NULL(stored_research)
 	return ..()
 
+/obj/machinery/mineral/processing_unit/multitool_act(mob/living/user, obj/item/multitool/M)
+	if(istype(M))
+		if(istype(M.buffer, /obj/machinery/mineral/processing_unit_console))
+			CONSOLE = M.buffer
```

Comment:
Out of scope (i.e. don't worry about this), but I really need to make the buffer only take WEAKREFs so that machines in buffers can properly QDEL